autogen/cluster.tf

```diff
@@ -225,62 +225,63 @@ resource "google_container_node_pool" "pools" {
   {% else %}
   provider = google
   {% endif %}
-  count    = length(var.node_pools)
-  name     = var.node_pools[count.index]["name"]
+  for_each = var.node_pools
```

Comment:
Let's keep the external interface as list but build the map internally within the module.

Reply:
Will keep the existing interface via having this in locals:
```tf
node_pool_names = [for np in toset(var.node_pools) : np.name]
node_pools = zipmap(local.node_pool_names, tolist(toset(var.node_pools)))
```